File: app/src/main/java/florent37/github/com/nosql/MainActivity.java
package florent37.github.com.nosql;

import android.app.Activity;
import android.os.Bundle;
import android.widget.TextView;

import com.github.florent37.androidnosql.Listener;
import com.github.florent37.androidnosql.NoSql;

import java.util.Arrays;

import florent37.github.com.nosql.model.Car;
import florent37.github.com.nosql.model.House;
import florent37.github.com.nosql.model.User;

public class MainActivity extends Activity {

    TextView textView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        textView = ((TextView) findViewById(R.id.text));

        final NoSql noSql = NoSql.getInstance();

        final User user = new User(
                "flo",
                new House("paris"),
                Arrays.asList(3, 5),
                Arrays.asList(new Car("chevrolet", "camaro"), new Car("ford", "gt"))
        );

        noSql.notify("/user/", new Listener() {
            @Override
            public void nodeChanged(String path, NoSql.Value value) {

            }
        });

        noSql.put("/user/florent/", user);

        final User userFetched = noSql.get("/user/florent/", User.class);
        textView.setText(userFetched.toString());
    }
}

File: app/src/main/java/florent37/github/com/nosql/MainActivity.kt
package florent37.github.com.nosql

import android.os.Bundle
import android.view.View
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import com.github.florent37.androidnosql.Listener
import com.github.florent37.androidnosql.NoSqlKt
import com.github.florent37.androidnosql.NosqlElement
import florent37.github.com.nosql.model.Car
import florent37.github.com.nosql.model.House
import florent37.github.com.nosql.model.User
import kotlinx.coroutines.*
import java.util.*

class MainActivity : AppCompatActivity(R.layout.activity_main), CoroutineScope by MainScope() {

    override fun onCreate(savedInstanceState: Bundle?) {

        super.onCreate(savedInstanceState)
        val user = User(
                "flo",
                House("paris"),
                listOf(3, 5),
                listOf(
                        Car("chevrolet", "camaro"),
                        Car("ford", "gt")
                )
        )

        val noSql = NoSqlKt
        launch {
            noSql.notify("/user/", object : Listener {
                override fun nodeChanged(path: String, element: NosqlElement) {

                }
            })
            noSql.put("/user/florent/", user)
            val userFetched: User? = noSql.get("/user/florent/", User::class.java)

            val textView : TextView = findViewById(R.id.text)
            textView.text = userFetched.toString()
        }
    }

    override fun onDestroy() {
        cancel()
        super.onDestroy()
    }
}